test/gtest/conv_test_base.hpp

```diff
@@ -211,3 +142,69 @@ struct ConvFwdSolverTestBase
     miopen::Allocator::ManageDataPtr wei_dev;
     miopen::Allocator::ManageDataPtr out_dev;
 };
+
+template <typename T, typename Tref, bool use_cpu_ref>
+void ConvFwdSolverTestBase<T, Tref, use_cpu_ref>::SetUpImpl(ConvTestCaseBase conv_config,
```

Comment:
Does the linker throw an error when building this code? Like "multiple definitions" or something?

Reply:
All multiple definitions errors have been resolved.